File: rb/features.rb
# Avatar SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module AvatarFeatures
  def self.make_feature(name)
    case name
    when "base"
      AvatarBaseFeature.new
    when "ratelimit"
      AvatarRatelimitFeature.new
    when "retry"
      AvatarRetryFeature.new
    when "test"
      AvatarTestFeature.new
    when "timeout"
      AvatarTimeoutFeature.new
    else
      AvatarBaseFeature.new
    end
  end
end

File: py/avatar_sdk/features.py
# Avatar SDK feature factory

from avatar_sdk.feature.base_feature import AvatarBaseFeature
from avatar_sdk.feature.test_feature import AvatarTestFeature


def _make_feature(name):
    features = {
        "base": lambda: AvatarBaseFeature(),
        "test": lambda: AvatarTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
